galata/test/jupyterlab/settings.test.ts

```diff
@@ -288,3 +288,49 @@ test('Keyboard Shortcuts: overwriting a shortcut can be cancelled', async ({
 
   await expect(conflict).toHaveCount(0);
 });
+
+test('Keyboard Shortcuts: validate "Or" button behavior when editing shortcuts', async ({
+  page
+}) => {
+  await page.evaluate(async () => {
+    await window.jupyterapp.commands.execute('settingeditor:open', {
+      query: 'Keyboard Shortcuts'
+    });
+  });
+
+  const shortcutsForm = page.locator('.jp-Shortcuts-ShortcutUI');
+  const filterInput = shortcutsForm.locator('jp-search.jp-FilterBox');
+  await filterInput.locator('input').fill('merge cell below');
+
+  const shortcutsContainer = page.locator(
+    '.jp-Shortcuts-ShortcutListContainer'
+  );
+  const firstRow = shortcutsContainer.locator('.jp-Shortcuts-Row').first();
+
+  const shortcutKeysCount = await firstRow
+    .locator('.jp-Shortcuts-ShortcutKeys')
+    .count();
+  const orCount = await firstRow.locator('.jp-Shortcuts-Or').count();
+  const orForcedCount = await firstRow
+    .locator('.jp-Shortcuts-Or-Forced')
+    .count();
+
+  expect(orCount).toBe(shortcutKeysCount);
+  expect(orForcedCount).toBe(shortcutKeysCount - 1);
+
+  const shortcutKey = firstRow.locator('.jp-Shortcuts-ShortcutKeys').first();
+  await shortcutKey.click();
+
+  // After clicking, assert that 'jp-Shortcuts-Or' count equals 'jp-Shortcuts-Or-Forced'
```

Comment:
Without looking at the code it is not clear to me why this should be the case. Ideally integration tests should be easy to understand and describe behaviour in a way independent of implementation details. The fact that the number of elements with these classes is equal seems disconnected from the behaviour being fixed here.

I think I would lean more towards a test case capturing a small screenshot of the relevant area as this is easy to validate.

Reply:
I focused on testing the exact behavior that was coded since it was my first time writing a proper visual test, so I approached it with more of a programmer mindset. I get what you're saying though—I'll aim to keep it simple and easily understandable.

Since this PR is about hiding the 'or' button and the 'Add' button during shortcut editing, how about I take a snapshot during the test and match it?"